src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -0,0 +1,836 @@
+import static uk.gov.hmcts.darts.arm.util.ArchiveConstants.ArchiveResponseFileAttributes.ARM_RESPONSE_SUCCESS_STATUS_CODE;
+import static uk.gov.hmcts.darts.arm.util.ArchiveConstants.ArchiveResponseFileAttributes.ARM_UPLOAD_FILE_FILENAME_KEY;
+import static uk.gov.hmcts.darts.arm.util.ArmResponseFilesUtil.generateSuffix;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_CHECKSUM_VERIFICATION_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_MANIFEST_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.ARM_RESPONSE_PROCESSING_FAILED;
+import static uk.gov.hmcts.darts.common.enums.ObjectRecordStatusEnum.STORED;
+
+@Slf4j
+@SuppressWarnings({"PMD.GodClass", "PMD.CyclomaticComplexity", "PMD.CouplingBetweenObjects"})
+public abstract class AbstractArmBatchProcessResponseFiles implements ArmResponseFilesProcessor {
+
+    protected static final String UNABLE_TO_UPDATE_EOD = "Unable to update EOD";
+    protected final ExternalObjectDirectoryRepository externalObjectDirectoryRepository;
+    protected final ArmDataManagementApi armDataManagementApi;
+    protected final FileOperationService fileOperationService;
+    protected final ArmDataManagementConfiguration armDataManagementConfiguration;
+    protected final ObjectMapper objectMapper;
+    protected final UserIdentity userIdentity;
+    protected final CurrentTimeHelper timeHelper;
+    protected final ExternalObjectDirectoryService externalObjectDirectoryService;
+    protected final Integer batchSize;
+    protected final LogApi logApi;
+
+    private UserAccountEntity userAccount;
+    private String continuationToken;
+
+    public AbstractArmBatchProcessResponseFiles(ExternalObjectDirectoryRepository externalObjectDirectoryRepository, ArmDataManagementApi armDataManagementApi,
+                                            FileOperationService fileOperationService, ArmDataManagementConfiguration armDataManagementConfiguration,
+                                            ObjectMapper objectMapper, UserIdentity userIdentity, CurrentTimeHelper timeHelper,
+                                            ExternalObjectDirectoryService externalObjectDirectoryService, Integer batchSize,
+                                            LogApi logApi) {
+        this.externalObjectDirectoryRepository = externalObjectDirectoryRepository;
+        this.armDataManagementApi = armDataManagementApi;
+        this.fileOperationService = fileOperationService;
+        this.armDataManagementConfiguration = armDataManagementConfiguration;
+        this.objectMapper = objectMapper;
+        this.userIdentity = userIdentity;
+        this.timeHelper = timeHelper;
+        this.externalObjectDirectoryService = externalObjectDirectoryService;
+        this.batchSize = batchSize;
+        this.logApi = logApi;
+    }
```

Comment:
Boiler plate could be removed with Lombok

Reply:
batch size is passed in to constructor so no lombok is not an option here

Reply:
That would be for the consuming classes not the abstract :) so I Lombok is still a valid scenario here

Lombok would not change core code logic just remove boiler plate which integration tests / automated tests should already cover